### DISCUSSION TITLE
All urls needed for clientid/secret/certificate auth flows?

### DISCUSSION THREAD

**Discussion by brianpmccullough** [pnp/pnpcore#1079] [Q&A]
Are all the urls needed for clientid/certificate auth flows documented/summarized somewhere?  In a locked down server enviornment I need to know exactly which domains to open up.  Thank you!

**Comment by jansenbe** [comment]
@brianpmccullough : they're not and they depend on the cloud environment you're in? The Azure AD auth and Microsoft Graph URLs differ between for example the regular production cloud and the US Government (GCC) cloud. See https://learn.microsoft.com/en-us/graph/deployments for more information
